### ISSUE TITLE
Update README to include secure best practices in handling secrets/creds in git repos

### ISSUE THREAD

**Issue by ScriptAutomate** [sphinx-contrib/confluencebuilder#442]
This sections about storing API keys/passwords in the `conf.py` file is a very bad idea when it comes to best practices in code repositories:

- https://github.com/sphinx-contrib/confluencebuilder#configuration
- https://sphinxcontrib-confluencebuilder.readthedocs.io/en/stable/configuration.html
  - Source: https://github.com/sphinx-contrib/confluencebuilder/blob/master/doc/configuration.rst

You should never store this information within a git repository, especially in plain-text. Your CI pipeline tooling should be handling secure content, so that any contributor isn't simply pulling down plain-text API keys to their workstation in order to just contribute to docs.

If using something like GitHub, and GitHub Actions, for example:
- https://docs.github.com/en/actions/reference/encrypted-secrets

GitLab CI docs can also be referenced for managed env var secrets, or external providers of secrets, if using their pipeline tooling.

For more information on why password storing in git is a bad idea, there is a good discussion and breakdown of things on Stack Exchange here:

- https://security.stackexchange.com/questions/191590/why-is-storing-passwords-in-version-control-a-bad-idea

**Comment by jdknight** [comment]
There is no intent to promote the idea that a user should store a plaintext password/secrets inside a shared version control system. The quick README/minimalistic configuration examples are there solely to help a new user of this extension to quickly setup a "first" environment. If users are actually storing the contents of `confluence_server_pass` in, for example, a shared Git repository, I would suspect there would be additional security concerns to consider for said users.

That being said, this is not the [first time](https://github.com/sphinx-contrib/confluencebuilder/issues/96) the best practices for secrets has been brought up. While a notice was added inside `confluence_server_pass`'s explicit section and the tip's document was updated on a way to deal with CI environments, maybe there can be more done here. For example:

- For the project README, tutorial example and the mini-configuration in the configuration document, we can replace `confluence_server_pass` with `confluence_ask_password`. When users then want to prepare to, for example, set a CI setup, they can find more details on `confluence_server_pass` when searching the documentation and be given the relevant warnings then. A concern to consider is trying to handle the use case where MinGW users can have trouble using the `confluence_ask_password` in their sessions; or,
- Leave the `confluence_server_pass` entries as is, but add a trailing comment such as the following when referenced:
```
confluence_server_pass = '<value>' # do not share secret
```
- Dedicate a section insider the "Tips" document which points a general description on secrets management, and have the `confluence_server_pass` configuration section point to this new section.
- Add a warning note and reference to new "Tips" section in the `confluence_client_cert_pass` option as well.

**Comment by jdknight** [comment]
With changes to the `confluence_ask_password` option to better supprt MSYS/MinGW environments (see #456), it was opted to adjust a series of documentation examples (see #462) to use the ask-password option over an explicit password configuration. From this, users who wish to investigation the ability to explicitly provide passwords for a publish run will lead the to the configuration-list document which includes an explicit caution about secrets management.

Note that changes to #462 do not yet include changes for the `README` document, which also includes an example configuration set. Changes to `README` are planned (with the following patch); however, they will be held until near a release event, to help reduce the chance of user's using the current stable release without the fixes in #456:
```
diff --git a/README.rst b/README.rst
index 940f8e8..3d58a0b 100644
--- a/README.rst
+++ b/README.rst
@@ -86,8 +86,8 @@ and publishing:
    confluence_space_name = 'TEST'
    confluence_parent_page = 'Documentation'
    confluence_server_url = 'https://intranet-wiki.example.com/'
-   confluence_server_user = 'username'
-   confluence_server_pass = 'api-key-or-password'
+   confluence_ask_user = True
+   confluence_ask_password = True
 
 For a complete list of configuration options, see also:
 
```

**Comment by jdknight** [comment]
The [above patch](https://github.com/sphinx-contrib/confluencebuilder/issues/442#issuecomment-826161779) has been [applied](https://github.com/sphinx-contrib/confluencebuilder/commit/c93dc18fb10abfab79a0fdb4b9a3ef66c57f24cd) to the main branch. This should finalize the changes desired to help reduce the risk of new users persisting secrets in various repositories. With v1.5 is now available on PyPI and the stable documentation now promoting ask options by default, marking as closed.
